src/storage/src/hummock/sstable/builder.rs

```diff
@@ -202,7 +202,10 @@ impl<W: SstableWriter> SstableBuilder<W> {
             // 1. not empty_key
             // 2. extract_key key is not duplicate
             if !extract_key.is_empty()
-                && (extract_key != &self.last_full_key[0..self.last_bloom_filter_key_length])
+                && (extract_key
+                    != &self.last_full_key[self.filter_key_extractor.start_index()
```

Comment:
In my opinion, this is incorrect, the filter_key_extractor belonging to the builder is a Multitype, which contains a variety of catalogs.

so the start_index is incorrect